components/blog-article.tpl

```diff
@@ -8,7 +8,9 @@
       {% endif %}
     </h1>
 
-    <time class="article__date" datetime="{{ article.created_at | date: '%Y-%m-%d' }}">{{ article.created_at | format_date: "long" }}</time>
+    {% assign current_year = 'now' | date: "%Y" | to_num %}
+    {% assign article_year = article.created_at | format_date: "%Y" | to_num %}
+    <time class="article__date" datetime="{{ article.created_at | date: '%Y-%m-%d' }}">{{ article.created_at | format_date: "long_without_year" }}{% if article_year < current_year %}, {{ article.created_at | format_date: "%Y" }}{% endif %}</time>
```

Comment:
Instead of extracting year number from the `article.created_at` value, use `{{ article.created_at | format_date: "long_without_year" }}` in one case and `{{ article.created_at | format_date: "long" }}` in the opposite case.

Also it might be a good idea to **assign** the correct format to a value in a separate if/else block for better readability.